docs/4.2/quickstart.md

```diff
@@ -164,10 +164,10 @@ accessible IP.
 
 !!! warning "Warning":
     For the purposes of this quickstart we are using the
-    `--insecure-no-tls` flag which allows us to skip configuring the HTTP/TLS
```

Comment:
I tested this flag out with tsh, and it didn't work, so I think it meant `--insecure` instead

Reply:
Yeah, this causes some confusion.

For `tsh`, the flag is `--insecure` to allow insecure connections.
For `teleport`, the flag is `--insecure-no-tls` to serve the web interface via HTTP rather than HTTPS.

Reply:
@webvictim Oh wow, that needs to be clarified:

> For the purposes of this quickstart we are using the `--insecure-no-tls` flag which allows us to skip configuring the HTTP/TLS

The quickstart did not use teleport `--insecure-no-tls`, so the above statement is false. Should I change the quickstart, to start teleport with `--insecure-no-tls`? https://gravitational.com/teleport/docs/quickstart/#step-2-start-teleport

Reply:
Yes, please do.

Reply:
@webvictim, I run into an issue when I `teleport start --insecure-no-tls`. It seems to start. 

But I run into problems:
- I can access the web UI without using https, but I cannot log in
- logging in through the CLI `tsh --proxy=kimlisa --insecure login`, gives me this error: 

```WARNING: You are using insecure connection to SSH proxy https://kimlisa:3080
error: Get https://kimlisa:3080/v1/webapi/ping: http: server gave HTTP response to HTTPS client```

Am I doing something wrong? Or is this a bug?

Reply:
So I think the issue is this:

`--insecure-no-tls` is designed for when people use Teleport behind load balancers which terminate SSL/TLS themselves - it means that the load balancer can pass web requests to Teleport without needing to use HTTPS itself. There is still always the expectation that the `tsh` client will use HTTPS to communicate with Teleport, though - and that the HTTPS needed here will be provided by the load balancer rather than Teleport itself.

`--insecure` means that the `tsh` client does not validate that the SSL/TLS certificate presented by Teleport is valid i.e. it has a correct hostname, is not an expired certificate, etc. When Teleport is started without a SSL/TLS certificate being specified (using the `https_key_file` and `https_cert_file` values in the Teleport config file) then it automatically generates a self-signed certificate, but this will not pass strict validation. So for quickstarts, we use the `--insecure` parameter to disable certificate validation and to allow the `tsh` client to connect.

Reply:
I think for now you should probably just merge this PR. I'll take a quick pass through the quickstart docs to remove the mentions of `--insecure-no-tls` and check that it makes sense.